docs/docs/creating-and-modifying-pages.md

```diff
@@ -92,13 +92,103 @@ exports.createPages = async ({ graphql, actions, reporter }) => {
       // In your blog post template's graphql query, you can use pagePath
       // as a GraphQL variable to query for data from the markdown file.
       context: {
-        pagePath: path,
-      },
+        pagePath: path
+      }
     })
   })
 }
 ```
 
+## Trade-offs of querying for all fields in the context object of `gatsby-node.js`
+
+Imagine a scenario where you could query for all the parameters your template would need in the `gatsby-node.js`. What are the performance advantages this method might propose? In this section, you will look into this.
+
+In the initial approach you have seen how the `gatsby-node.js` file would have a query block like so :
+
+```graphql
+  const queryResults = await graphql(`
+    query AllProducts {
+      AllProducts {
```

Comment:
```suggestion
      allProducts {
```

Seems like Gatsby defaults to camelCase on these aggregate GraphQL fields when they're generated.